presto-plugin-toolkit/src/main/java/io/prestosql/plugin/base/security/FileBasedSystemAccessControl.java

```diff
@@ -334,6 +338,26 @@ public void checkCanDropSchema(SystemSecurityContext context, CatalogSchemaName
         if (!canAccessCatalog(context.getIdentity(), schema.getCatalogName(), ALL)) {
             denyDropSchema(schema.toString());
         }
+
+        if (!isSchemaOwner(context, schema.getSchemaName())) {
+            denyDropSchema(schema.getSchemaName());
+        }
+    }
+
+    private boolean isSchemaOwner(SystemSecurityContext context, String schemaName)
```

Comment:
please move this method somewhere to the bottom of this class. Favor public interface first and also order methods by usage.